data/blog/session-replay-sdk-bundle-size-optimizations.md

```diff
@@ -0,0 +1,209 @@
+
+In version 7.73.0 of the JavaScript SDKs, we updated the underlying [rrweb](https://github.com/getsentry/rrweb) package from v1 to v2.
+While this brought a host of improvements, it also came with a considerable increase in bundle size.
+This tipped us over the edge to declare a bundle size emergency, and focus on bringing the additional size Session Replay adds to the SDK down as much as possible.
+
+We're very happy to say that our efforts have been successful, and we managed to reduce the minified & gzipped bundle size compared to the rrweb 2 baseline by 23% (~19 KB), and by up to 35% (~29 KB) with maximum tree shaking configuration enabled.
+
+| Version                                                                      | Bundle Size¹ | What                                     |
+| ---------------------------------------------------------------------------- | ------------ | ---------------------------------------- |
+| [7.72.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.72.0) | 75.58 KB     | Before updating to rrweb 2.0             |
+| [7.73.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.73.0) | 84.26 KB     | After updating to rrweb 2.0              |
+| [7.78.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.78.0) | 65.24 KB     | New default                              |
+| [7.78.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.78.0) | 55.48 KB     | With all tree shaking options configured |
+
+¹Including Error & Performance Monitoring as well as Session Replay, minified & gzipped
+
+## Steps to reduce the SDK bundle size
+
+In order to achieve these bundle size improvements, we took a couple of steps ranging from removing unused code to build time configuration and improved tree shaking:
+
+- Allow to remove iframe & shadow DOM support via a build-time flag
+- Removed canvas recording support by default (users can opt-in via a config option, [support is coming](https://github.com/getsentry/sentry-javascript/issues/6519))
+- Remove unused code from our rrweb fork
+- Remove unused code in Session Replay itself
+- Allow to remove the included compression worker in favor of hosting it yourself
+- Move to a different compression library with a smaller footprint
+
+## Primer: rrweb
+
+[rrweb](https://github.com/getsentry/rrweb) is the underlying tool we use to make the recordings for Session Replay.
+While we try to contribute to the main rrweb repository as much as possible, there are some changes that are very specific to our needs at Sentry,
+which is why we also maintain a [forked version](https://github.com/getsentry/rrweb) of rrweb with some custom changes.
+
+## Primer: Tree Shaking
+
+If you do not know what tree shaking is and how it works, you can [read about it in our docs](https://docs.sentry.io/platforms/javascript/configuration/tree-shaking/).
```

Comment:
Actually I feel we need to give them a bit more here. Since we're adding a heading for the subject at least a tl;dr would be nice:

```suggestion
Tree shaking allows a JavaScript bundler to remove unused code from the final bundle
If you're not familiar with how it works and their advantages, we recommend [reading about it in our docs](https://docs.sentry.io/platforms/javascript/configuration/tree-shaking/) for more details.
```